packages/tupaia-web/src/utils/useMapOverlays.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+import { ChangeEvent } from 'react';
+import { useSearchParams, useParams } from 'react-router-dom';
+import { UseQueryResult } from 'react-query';
+import { useFetchMapOverlays } from '../api/queries';
+import { URL_SEARCH_PARAMS } from '../constants';
+import { MapOverlayGroup, SingleMapOverlayItem } from '../types';
+
+const flattenMapOverlays = (mapOverlayGroups: MapOverlayGroup[] = []): SingleMapOverlayItem[] => {
```

Comment:
Could we make this `mapOverlaysByCode` and have it generate a dictionary of the map overlays like below. Then we could just lookup the result below instead of finding it. eg `mapOverlaysByCode[selectedOverlayCode]`
``` 
{ 
  123: {...},
  128: {...}
}